inc/core/admin.php

```diff
@@ -70,16 +70,39 @@ function () {
 
 		add_filter( 'all_plugins', array( $this, 'change_plugin_names' ) );
 
-		add_action( 'after_switch_theme', array( $this, 'migrate_options' ) );
+		$this->auto_update_skin_and_builder();
 
-		$this->run_skin_and_builder_switches();
+		add_action( 'after_switch_theme', array( $this, 'migrate_options' ) );
 
 		add_filter( 'ti_tpc_theme_mods_pre_import', [ $this, 'migrate_theme_mods_for_new_skin' ] );
 
 		add_action( 'rest_api_init', [ $this, 'register_rest_routes' ] );
 		add_filter( 'neve_pro_react_controls_localization', [ $this, 'adapt_conditional_headers' ] );
 	}
 
+	/**
+	 * Automatic upgrade from legacy builder and skin on init.
+	 *
+	 * @return void
+	 */
+	private function auto_update_skin_and_builder() {
+		// If already on new skin bail.
+		if ( neve_was_auto_migrated_to_new() || get_theme_mod( 'neve_new_skin' ) === 'new' ) {
+			return;
+		}
+		set_theme_mod( 'neve_auto_migrated_to_new_skin', true );
+
+		$this->run_skin_and_builder_switches();
+
+		$migrator = new Builder_Migrator();
+		$response = $migrator->run();
+
+		if ( $response === true ) {
+			set_theme_mod( 'neve_migrated_builders', true );
```

Comment:
I saw other flags inside Neve's code, possibly related to the migration. I don't know what they do. I saw them in an e2e json. I think it's worth checking them, seeing what they do, and deciding if we need them or not.

```
 "neve_ran_builder_migration": true,
  "neve_migrated_builders": true,
  "neve_ran_migrations": true,
  "neve_had_old_skin": true,
  "neve_migrated_hfg_colors": true,
  "neve_new_skin": "new",
```

Reply:
The flags that I used are the only ones that are required for the migration modal to be presented in case it was not migrated and when doing the auto migration.

Reply:
@preda-bogdan I was suggesting checking those flags and seeing what they do. Maybe there is some more code that we don't need and can be removed.